model/src/main/kotlin/Provenance.kt

```diff
@@ -95,12 +93,11 @@ private class ProvenanceDeserializer : StdDeserializer<Provenance>(Provenance::c
             }
             node.has("vcs_info") -> {
                 val vcsInfo = jsonMapper.treeToValue<VcsInfo>(node["vcs_info"])!!
-                RepositoryProvenance(vcsInfo)
+                // For backward compatibility, if there is no resolved_revision use the revision from vcsInfo.
+                val resolvedRevision = node["resolved_revision"]?.textValue() ?: vcsInfo.revision
```

Comment:
Why is it necessary to fallback to the `revision`? The `match()` function before never matched if there is not `resolvedRevision` and due to this fallback it would match.

```
        // If no VCS information is present either, or it does not have a resolved revision, there is no way of
        // verifying matching provenance.
        if (vcsInfo.resolvedRevision == null) return false
```

Reply:
That's right, but in the backward compatibility case the deserialization of `VcsInfo` also prefers `resolvedRevision` over `revision`, that's why we have to fallback to `revision` to get the resolved revision of the scan result. Like I mentioned above, this does not always work which is why some packages will have to be rescanned, but at least it works in most cases.